### ISSUE TITLE
not more than one structural directive on one element

### ISSUE THREAD

**Issue by zoechi** [dart-lang/angular_analyzer_plugin#114]
This is invalid

```dart
<div *ngFor="let x of y" *ngIf="x.z == true"></div>
```

I haven't found another related issue but might be covered already.

**Comment by MichaelRFairhurst** [comment]
Thanks Gunter, this is definitely a great edge case to catch in the IDE. Always glad to get your input!

We have a rewrite of the angular template parser as its own library, I know that that parser catches this and probably a lot of other things we aren't catching yet. I'm hoping to move to that before too long to get all that goodness. (I should probably make a ticket for that)

If we that becomes blocked or delayed by enough time, we probably should code this up ourselves since its quick and useful.

**Comment by zoechi** [comment]
Glad to hear you find my comments useful.
It will be such a tremendous improvement for productivity when this becomes available.
Any timeframe (days, months, years, ... ;-) ) when this might be usable in a "normal" development environment. Seems there is still no plugin mechanism in place in the analyzer.

**Comment by MichaelRFairhurst** [comment]
closing this as it will be reported by angular_ast very soon
